src/gen_prog_align.jl

```diff
@@ -0,0 +1,664 @@
+# Kobe De Temmerman
+module GenProgAlign
+
+# Importing required packages
+using BioSequences, BioAlignments, StatsBase, Random
+# Export functions relevant for the user
+export  ga_alignment, 
+        fasta_to_array, 
+        Alignment, 
+        AA,
+        pairwise_score,
+        objective_function,
+        one_point_crossover,
+        gap_insertion,
+        shifting_one_seq,
+        shifting_block
+
+"""
+## ga_alignment
+    ga_alignment(path_to_fasta_file; population_size = 50, p = 0.95, max_operations_without_improvement = 2000, max_operations = 10000,score_matrix = BLOSUM62, gap_opening = -10, gap_extension = -1, symbols = AA)
+
+Uses a genetic algorithm to perform alignment between a set of sequences. 
+4 mutations are included:
+- Single point crossover: Left part of 1 parent alignment is combined with right part of other parent alignment
+- Gap insertion: At a random position of a random sequence in the alignment, one or more gaps are inserted.
+- Shifting one sequence: A part of one sequence in the alignment is shifted to the left or the right
+- Shifting a block: A contiguous block of residues is shifted in the alignment.
+
+The algorithm starts from a random population of allignments. Every generation the 50% individuals with the best
+alignment score  are kept the other 50% are replaced. To find replacements parent alignments
+are mutated and selected if the mutant has a higher alignment score than it's parent. 
+A dynamic schedule is used to select the consecutive operations, i.e. the next operation is chosen stochastically
+with a chance that depends on the recent performance history of the operations.
+
+Argument list
+- path\\_to\\_fasta_file: Path to fasta file containing the sequences to be aligned.
+- population\\_size: Size of the population, remains constant over generations.
+- p: Chance of inserting a gap during creation initial alignment. Necessary for Alignment function defined lower.
+- max\\_operations\\_without\\_improvement: Stop criterion, maximum consecutive fails to improve the score of the parent alignment with an operation.
+- max\\_operations: Stop criterion, total maximum operations performed. 
+- score\\_matrix: Score matrix used to calculate the alignment score, necessary for objective\\_function => pairwise\\_score defined lower. 
+- gap\\_opening: Penalty to introduce a gap in an alignment, necessary for objective\\_function => pairwise\\_score defined lower.
+- gap\\_extension: Penalty to extend a gap in an alignment, necessary for objective\\_function => pairwise\\_score defined lower.
+- symbols: Column and rownames of the score\\_matrix.
+"""
+function ga_alignment(path_to_fasta_file; population_size = 50, p = 0.95, max_operations_without_improvement = 2000, max_operations = 10000,score_matrix = BLOSUM62, gap_opening = -10, gap_extension = -1, symbols = AA)
+    ## Read in fastafile with  GenProgAlign.fasta_to_array 
```

Comment:
This and other comments are going too far, the code you're commenting is already self-explanatory. Reserve the comments for the cases when you can't express the intention with just good variable and function names.